phantom-drift/introspection.proto

```diff
@@ -0,0 +1,387 @@
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package introspection.pb;
+
+// ResultCounter is a monotonically increasing counter that reports an ok/err breakdown of the total.
+message ResultCounter {
+    uint32 total = 1;
+    uint32 ok = 2;
+    uint32 err = 3;
+}
+
+// Moving totals over sliding time windows. Models sensible time windows,
+// we don't have to populate them all at once.
+//
+// Graphical example:
+//
+// time     past -> present                              an event 16 min ago
+// ======================================================X================>>
+//                                                       |               | 1m
+//                                                       |           |---| 5m
+//                                                       | |-------------| 15m
+//                                          |------------X---------------| 30m
+//            |------------------------------------------X---------------| 60m
+message SlidingCounter {
+    uint32 over_1m      = 1;
+    uint32 over_5m      = 2;
+    uint32 over_15m     = 3;
+    uint32 over_30m     = 4;
+    uint32 over_1hr     = 5;
+    uint32 over_2hr     = 6;
+    uint32 over_4hr     = 7;
+    uint32 over_8hr     = 8;
+    uint32 over_12hr    = 9;
+    uint32 over_24hr    = 10;
+}
+
+// DataGauge reports stats for data traffic in a given direction.
+message DataGauge {
+    // Cumulative bytes.
+    uint64 cum_bytes    = 1;
+    // Cumulative packets.
+    uint64 cum_packets  = 2;
+    // Instantaneous bandwidth measurement.
+    uint64 inst_bw      = 3;
+}
+
+// Transport represents the state of a transport. Stats for the transport are
+// hosted and reported from the relevant component.
+message Transport {
+    // the id of this transport for referencing purposes.
+    bytes id       = 1;
+    // the name of this transport.
+    string name     = 2;
+    // whether this transport is enabled for listening.
+    bool listening  = 3;
+    // whether this transport is enabled for dialing.
+    bool dialing    = 4;
+    // the multiaddrs this transport is listening on, if enabled for listening.
+    repeated string listen_multiaddrs = 5;
+}
+
+// Dialer encapsulates stats and state about the swarm dialer.
+message Dialer {
+    // InflightDial represents a dial that is in progress.
+    message InflightDial {
+        string peer_id          = 1;
+        uint32 addrs_cnt        = 2;
```

Comment:
Is `addrs_cnt` the number of addresses that we are dialing as part of this same dial attempt? Or the number of addresses that the peer has?

Reply:
The number of addresses we've decided to dial for that peer. This can increase/decrease as we discover more addresses while the dial is inflight, or we decide to filter out addresses dynamically.